### PR TITLE
Debug commands for regions

### PR DESCRIPTION
**Pull request by jmpesp**
Adds the following debug commands:

- list all regions
- given a list of regions, find what is using that region
- find all deleted volume regions

This was carved out of oxidecomputer/omicron#4378.

### CODE REVIEW

**Comment by leftwo** [comment]
I love it.

Could you included some sample output from the new commands so we can bikeshed it?
I would love to bikeshed some formatting, but I'm too lazy to clone/compile/run the code.

File: dev-tools/omdb/src/bin/omdb/db.rs [hunk 1]
```diff
@@ -1994,6 +2040,305 @@ async fn cmd_db_region_missing_porst(
...
+            .to_string();
+
+        println!("{}", table);
+    }
+
+    Ok(())
+}
+
+/// Find what is using a region
+async fn cmd_db_region_used_by(
```
Comment (leftwo) [inline dev-tools/omdb/src/bin/omdb/db.rs:2104]: This seems really cool, but.. does it just take any old UUID and...  how does it work?
Are we providing the UUID of the region in `RegionUsedByArgs`?

Comment (jmpesp) [reply dev-tools/omdb/src/bin/omdb/db.rs:2104]: We are yeah:
```
$ ./target/debug/omdb db region list
 id                                   | dataset_id                           | volume_id                            | block_size | blocks_per_extent | extent_count | read_only 
--------------------------------------+--------------------------------------+--------------------------------------+------------+-------------------+--------------+-----------
 27eaaabc-ab7e-4d3a-bf54-e1bbc86384f5 | bb7b9ada-9aab-4a9d-be89-23ddd09a29d2 | aa54c368-eee0-4e9b-86a9-58b1630da4bb | 512        | 131072            | 32           | false     
 3f7e5898-ec73-48bc-86a0-ef7a998b87db | bb7b9ada-9aab-4a9d-be89-23ddd09a29d2 | ffdae556-6b03-4372-ba15-d2e8ad9fa1c1 | 512        | 131072            | 32           | false     
 42e49268-f0a7-431e-999e-3e82d03df045 | c728295a-97bc-4d42-bc31-f29b5c721208 | ffdae556-6b03-4372-ba15-d2e8ad9fa1c1 | 512        | 131072            | 32           | false     
 81f077e0-346d-46f8-bdd6-b2f84ad3f34e | 70118760-7acd-4d8a-9fe7-7cd2d3c8e72d | aa54c368-eee0-4e9b-86a9-58b1630da4bb | 512        | 131072            | 32           | false     
 90f98fe7-011b-4a16-9378-362d113ef245 | 70118760-7acd-4d8a-9fe7-7cd2d3c8e72d | ffdae556-6b03-4372-ba15-d2e8ad9fa1c1 | 512        | 131072            | 32           | false     
 a7725c22-cda9-4f30-b3dd-deadd94d812b | c728295a-97bc-4d42-bc31-f29b5c721208 | aa54c368-eee0-4e9b-86a9-58b1630da4bb | 512        | 131072            | 32           | false     

$ ./target/debug/omdb db region used-by 27eaaabc-ab7e-4d3a-bf54-e1bbc86384f5
 id                                   | volume_id                            | usage_type | usage_id                             | usage_name           | deleted 
--------------------------------------+--------------------------------------+------------+--------------------------------------+----------------------+---------
 27eaaabc-ab7e-4d3a-bf54-e1bbc86384f5 | aa54c368-eee0-4e9b-86a9-58b1630da4bb | disk       | 1b39bc7d-2ef2-434d-a8eb-0149e437d6fe | crucible-tester-disk | false
 ```

**Comment by jmpesp** [comment]
> I love it.
> 
> Could you included some sample output from the new commands so we can bikeshed it? I would love to bikeshed some formatting, but I'm too lazy to clone/compile/run the code.

definitely!

list all regions:

```
$ ./target/debug/omdb db region list 2>/dev/null
 id                                   | dataset_id                           | volume_id                            | block_size | blocks_per_extent | extent_count | read_only 
--------------------------------------+--------------------------------------+--------------------------------------+------------+-------------------+--------------+-----------
 27eaaabc-ab7e-4d3a-bf54-e1bbc86384f5 | bb7b9ada-9aab-4a9d-be89-23ddd09a29d2 | aa54c368-eee0-4e9b-86a9-58b1630da4bb | 512        | 131072            | 32           | false     
 3f7e5898-ec73-48bc-86a0-ef7a998b87db | bb7b9ada-9aab-4a9d-be89-23ddd09a29d2 | ffdae556-6b03-4372-ba15-d2e8ad9fa1c1 | 512        | 131072            | 32           | false     
 42e49268-f0a7-431e-999e-3e82d03df045 | c728295a-97bc-4d42-bc31-f29b5c721208 | ffdae556-6b03-4372-ba15-d2e8ad9fa1c1 | 512        | 131072            | 32           | false     
 81f077e0-346d-46f8-bdd6-b2f84ad3f34e | 70118760-7acd-4d8a-9fe7-7cd2d3c8e72d | aa54c368-eee0-4e9b-86a9-58b1630da4bb | 512        | 131072            | 32           | false     
 90f98fe7-011b-4a16-9378-362d113ef245 | 70118760-7acd-4d8a-9fe7-7cd2d3c8e72d | ffdae556-6b03-4372-ba15-d2e8ad9fa1c1 | 512        | 131072            | 32           | false     
 a7725c22-cda9-4f30-b3dd-deadd94d812b | c728295a-97bc-4d42-bc31-f29b5c721208 | aa54c368-eee0-4e9b-86a9-58b1630da4bb | 512        | 131072            | 32           | false     
 ```

find what a region is used by:

```
$ ./target/debug/omdb db region used-by 27eaaabc-ab7e-4d3a-bf54-e1bbc86384f5 2>/dev/null
 id                                   | volume_id                            | usage_type | usage_id                             | usage_name           | deleted 
--------------------------------------+--------------------------------------+------------+--------------------------------------+----------------------+---------
 27eaaabc-ab7e-4d3a-bf54-e1bbc86384f5 | aa54c368-eee0-4e9b-86a9-58b1630da4bb | disk       | 1b39bc7d-2ef2-434d-a8eb-0149e437d6fe | crucible-tester-disk | false
```

output the list of all regions, and find what they're all used by:

```
$ ./target/debug/omdb db region list -i 2>/dev/null
27eaaabc-ab7e-4d3a-bf54-e1bbc86384f5
3f7e5898-ec73-48bc-86a0-ef7a998b87db
42e49268-f0a7-431e-999e-3e82d03df045
81f077e0-346d-46f8-bdd6-b2f84ad3f34e
90f98fe7-011b-4a16-9378-362d113ef245
a7725c22-cda9-4f30-b3dd-deadd94d812b

$ ./target/debug/omdb db region list -i 2>/dev/null | xargs ./target/debug/omdb db region used-by 2>/dev/null
 id                                   | volume_id                            | usage_type    | usage_id                             | usage_name               | deleted 
--------------------------------------+--------------------------------------+---------------+--------------------------------------+--------------------------+---------
 27eaaabc-ab7e-4d3a-bf54-e1bbc86384f5 | aa54c368-eee0-4e9b-86a9-58b1630da4bb | disk          | 1b39bc7d-2ef2-434d-a8eb-0149e437d6fe | crucible-tester-disk     | false   
 3f7e5898-ec73-48bc-86a0-ef7a998b87db | ffdae556-6b03-4372-ba15-d2e8ad9fa1c1 | snapshot dest | 283bfe6d-58cf-4cb6-9519-f8a2620f60d5 | crucible-tester-snapshot | false   
 42e49268-f0a7-431e-999e-3e82d03df045 | ffdae556-6b03-4372-ba15-d2e8ad9fa1c1 | snapshot dest | 283bfe6d-58cf-4cb6-9519-f8a2620f60d5 | crucible-tester-snapshot | false   
 81f077e0-346d-46f8-bdd6-b2f84ad3f34e | aa54c368-eee0-4e9b-86a9-58b1630da4bb | disk          | 1b39bc7d-2ef2-434d-a8eb-0149e437d6fe | crucible-tester-disk     | false   
 90f98fe7-011b-4a16-9378-362d113ef245 | ffdae556-6b03-4372-ba15-d2e8ad9fa1c1 | snapshot dest | 283bfe6d-58cf-4cb6-9519-f8a2620f60d5 | crucible-tester-snapshot | false   
 a7725c22-cda9-4f30-b3dd-deadd94d812b | aa54c368-eee0-4e9b-86a9-58b1630da4bb | disk          | 1b39bc7d-2ef2-434d-a8eb-0149e437d6fe | crucible-tester-disk     | false
```

**Comment by jmpesp** [comment]
`find-deleted-volume-regions` (regions that haven't been cleaned up yet) will only show something if the volume delete is interrupted:

```
$ oxide disk import ... --image "crucible-tester" ...

// after a disk import, (depending on the arguments) there's a disk, snapshot, 
// and image. there will be one region snapshot per region, and the snapshot   
// volume and image volume will have references to them.

// the disk delete will not actually delete the regions, as they have
// associated region snapshots.
$ oxide disk delete --project myproj --disk crucible-tester-disk

// neither will the snapshot delete, because the image still has references to 
// the region snapshots
$ oxide snapshot delete --project myproj --snapshot crucible-tester-snapshot   

// before the next command, disable a crucible agent service that is in a      
// dataset with one of the region snapshots
root@oxz_crucible_c728295a:~# svcadm disable svc:/oxide/crucible/agent:default 

// calling this this will then hang the volume delete saga
$ oxide image delete --project myproj --image crucible-tester

// now, `find-deleted-volume-regions` will work, showing which regions need to 
// be cleaned up:
$ ./target/debug/omdb db region find-deleted-volume-regions 2>/dev/null
 dataset_id                           | region_id                            | region_snapshot_id                   | volume_id
--------------------------------------+--------------------------------------+--------------------------------------+--------------------------------------
 13463660-d68b-47c3-bbf6-dfe2ea743b62 | 938609f4-d650-40f7-b63a-8d61dd9117af | 45b0d32b-392e-444b-8663-cb83e0959a4c | 3e195630-6a3a-419c-a73d-6c3249396c1c
 70118760-7acd-4d8a-9fe7-7cd2d3c8e72d | 979e381a-6178-4013-aafe-33762920a9ee | 45b0d32b-392e-444b-8663-cb83e0959a4c | 3e195630-6a3a-419c-a73d-6c3249396c1c
 c728295a-97bc-4d42-bc31-f29b5c721208 | 08cd3f59-7c4e-4cb9-b59b-f713033d3835 | 45b0d32b-392e-444b-8663-cb83e0959a4c | 3e195630-6a3a-419c-a73d-6c3249396c1c
```

**Review by leftwo — APPROVED** [review]

**Pull request merged**